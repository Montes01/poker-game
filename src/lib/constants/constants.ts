export const COMPANY_NAME = 'pragma';

export const cards = [
    { content: "0", voted: false },
    { content: "1", voted: false },
    { content: "2", voted: false },
    { content: "3", voted: false },
    { content: "4", voted: false },
    { content: "5", voted: false },
    { content: "6", voted: false },
    { content: "7", voted: false },
    { content: "8", voted: false },
    { content: "9", voted: false },
    { content: "â˜•", voted: false },
    { content: "?", voted: false },
];
