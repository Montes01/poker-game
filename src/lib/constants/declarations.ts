export enum playerType {
  player = "player",
  spectator = "spectator",
}
export interface player {
  name: string
  id: string
  type: keyof typeof playerType
  vote: string
}

export interface room {
  id: string
  name: string
  admin: string
  players: player[]
}

export interface Card {
  content: string
  voted: boolean
  count?: number
}
export enum ioEvents {
  createRoom = "createRoom",
  addPlayer = "addPlayer",
  vote = "vote",
  reset = "reset",
  reveal = "reveal",
  disconnect = "disconnect",
  connect = "connect",
  joinRoom = "joinRoom",
}