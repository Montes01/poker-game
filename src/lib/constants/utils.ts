import { cardTypes, Card } from "./declarations"

//must be between 5-20 characters
const LENGTH_REGEX = /^.{5,20}$/
//must contain less than 3 numbers
const NUMBER_REGEX = /^(?:\D*\d){0,3}\D*$/
//must not contain only numbers
const ONLY_NUMBER_REGEX = /[a-zA-Z]/
//must not contain special characters
const SPECIAL_CHAR_REGEX = /^[a-zA-Z0-9]*$/

//create the validator
export const validateName = (value: string) => {
  if (!LENGTH_REGEX.test(value)) {
    throw new Error("Debe tener entre 5 y 20 caracteres")
  }
  if (!ONLY_NUMBER_REGEX.test(value)) {
    throw new Error("No debe contener solo números")
  }
  if (!NUMBER_REGEX.test(value)) {
    throw new Error("Debe contener menos de 3 números")
  }
  if (!SPECIAL_CHAR_REGEX.test(value)) {
    throw new Error("No debe contener caracteres especiales")
  }
}

export const getFirstUserLetters = (name: string) => {
  if (name.length < 3) return name
  const words = name.split(" ")
  if (words.length > 1) return (words[0][0] + words[1][0]).toUpperCase()
  else return (words[0][0] + words[0][1]).toUpperCase()
}

export function generateLink(id:string) {
  let link = ""
  if (typeof window !== "undefined") {
    link = `${window.location.origin}/room/${id}`
  } else {
    link = `http://localhost/room/${id}`
  }
  return link
}

export function generateCards(type: keyof typeof cardTypes) {
  const cards: Card[] = []
  if (type === "fibonacci") {
    cards.push({ content: "0", voted: false })
    cards.push({ content: "1", voted: false })
    cards.push({ content: "2", voted: false })
    for (let i = 2; i < 9; i++) {
      cards.push({
        content: `${Number(cards[i].content) + Number(cards[i - 1].content)}`,
        voted: false,
      })
    }
  } else if (type === "normal") {
    for (let i = 0; i < 10; i++) {
      cards.push({ content: `${i}`, voted: false })
    }
  } else if (type === "tenX") {
    for (let i = 0; i < 10; i++) {
      cards.push({ content: `${i * 10}`, voted: false })
    }
  }
  cards.push({ content: "☕", voted: false })
  cards.push({ content: "?", voted: false })
  return cards
}