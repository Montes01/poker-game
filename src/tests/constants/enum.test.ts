import {playerType, inputTypes, ioEvents, } from "../../lib/constants/declarations"

describe("enum", () => {
  it("should match the snapshot", () => {
    expect(playerType).toMatchSnapshot()
    expect(inputTypes).toMatchSnapshot()
    expect(ioEvents).toMatchSnapshot()
  })
})
