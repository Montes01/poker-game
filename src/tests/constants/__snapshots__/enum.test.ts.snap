// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enum should match the snapshot 1`] = `
{
  "player": "player",
  "spectator": "spectator",
}
`;

exports[`enum should match the snapshot 2`] = `
{
  "checkbox": "checkbox",
  "color": "color",
  "date": "date",
  "email": "email",
  "file": "file",
  "month": "month",
  "number": "number",
  "password": "password",
  "radio": "radio",
  "range": "range",
  "search": "search",
  "tel": "tel",
  "text": "text",
  "time": "time",
  "url": "url",
  "week": "week",
}
`;

exports[`enum should match the snapshot 3`] = `
{
  "addPlayer": "addPlayer",
  "changeCards": "changeCards",
  "changeType": "changeType",
  "connect": "connect",
  "createRoom": "createRoom",
  "giveAdmin": "giveAdmin",
  "joinRoom": "joinRoom",
  "reset": "reset",
  "reveal": "reveal",
  "updateRoom": "updateRoom",
  "vote": "vote",
}
`;
